mm2src/db_common/src/async_sql_conn.rs

```diff
@@ -0,0 +1,288 @@
+use crate::sqlite::rusqlite::Error as SqlError;
+use crossbeam_channel::Sender;
+use futures::channel::oneshot::{self};
+use rusqlite::OpenFlags;
+use std::fmt::{self, Debug, Display};
+use std::path::Path;
+use std::thread;
+
+/// Represents the errors specific for AsyncConnection.
+#[derive(Debug)]
+pub enum AsyncConnError {
+    /// The connection to the SQLite has been closed and cannot be queried any more.
+    ConnectionClosed,
+    /// An error occurred while closing the SQLite connection.
+    /// This `Error` variant contains the [`AsyncConnection`], which can be used to retry the close operation
+    /// and the underlying [`SqlError`] that made it impossible to close the database.
+    Close((AsyncConnection, SqlError)),
+    /// A `Rusqlite` error occurred.
+    Rusqlite(SqlError),
+    /// An application-specific error occurred.
+    Other(Box<dyn std::error::Error + Send + Sync + 'static>),
```

Comment:
I don't think we need to have this much dynamic type on mm2 which makes `Box<dyn` an overkill for us. Can we use simpler type instead? Even something like `Internal(String)` should be fine I guess?

Reply:
yeah, agree. added `Internal(InternalError)` where inner type is `pub struct InternalError(pub String);`